src/main/kotlin/uk/gov/dluhc/registercheckerapi/messaging/MessageQueue.kt

```diff
@@ -1,11 +1,23 @@
 package uk.gov.dluhc.registercheckerapi.messaging
 
 import io.awspring.cloud.messaging.core.QueueMessagingTemplate
+import org.springframework.messaging.Message
+import org.springframework.messaging.core.MessagePostProcessor
 
 class MessageQueue<T : Any>(
     private val queueName: String,
     private val queueMessagingTemplate: QueueMessagingTemplate
 ) {
-    fun submit(message: T) =
-        queueMessagingTemplate.convertAndSend(queueName, message)
+    fun submit(payload: T) =
+        queueMessagingTemplate.send(queueName, queueMessagingTemplate.convertToMessage(payload))
```

Comment:
@NathanRussellValtech ,

Same question as the `VCA` one.

I realized that we have a convertAndSend method in QueueMessagingTemplate class which internally also calls doConvert method as our external function does.

Can that method be used instead for simplicity?

fun submit(payload: T) = queueMessagingTemplate.convertAndSend(queueName, payload)

Reply:
As discussed on slack, we are trying to ensure that `org.springframework.messaging.core.AbstractMessageChannelMessagingSendingTemplate.send(String queueName, Message message)` is invoked, which is the only public method that we can weave that takes a `Message` as one of it's arguments, so that the aspect can manipulate the `Message` by adding a correlationID header to it.

`queueMessagingTemplate.convertAndSend(queueName, payload)` invokes a subtly different call stack, where the first public method that takes a `Message` as an argument is `org.springframework.messaging.core.AbstractMessageSendingTemplate.send(D destimation, Message message)` which does not get weaved over for some reason